src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -293,6 +298,50 @@ private function autowireMethod(\ReflectionMethod $reflectionMethod, array $argu
     }
 
     /**
+     * Autowires getters.
+     *
+     * @param array $overridenGetters
+     * @param array $autowiredMethod
+     *
+     * @return array
+     */
+    private function autowireOverridenGetters(array $overridenGetters, array $autowiredMethod)
+    {
+        foreach ($autowiredMethod as $reflectionMethod) {
+            if (isset($overridenGetters[strtolower($reflectionMethod->name)])
+                || !method_exists($reflectionMethod, 'getReturnType')
+                || 0 !== $reflectionMethod->getNumberOfParameters()
+                || $reflectionMethod->isFinal()
+                || $reflectionMethod->returnsReference()
+                || !($returnType = $reflectionMethod->getReturnType())
```

Comment:
I can partially answer here: this list is just the lits of cases where explicit getter injection throws an exception already - thus cases where autowiring should just skip.
A comment would be nice for sure, but splitting the condition is not needed imho.